src/components/TodoForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import usersFromServer from '../api/users';
+import { TodosWithUsers } from '../type.ts/TodosWithUser';
+import { getUserById } from '../utils/userId';
+
+type Props = {
+  onAdd: (todoWithUser: TodosWithUsers) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const onUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
```

Comment:
Ensure that `getUserById(userId)` returns a valid user object. If it returns `null`, handle this case appropriately to avoid issues when adding a todo.